WIP PR for BXMSDOC-6281: Update the managing & monitoring KIE Server document as per RHPAM/RHDM 7.9 release. (#93)
kaldesai/kie-docs: doc-content/enterprise-only/installation/kie-server-configure-central-proc.adoc

[id='kie-server-configure-central-proc_{context}']
= Configuring {KIE_SERVER} to connect to {CENTRAL}

WARNING: This section provides a sample setup that you can use for testing purposes. Some of the values are unsuitable for a production environment, and are marked as such.

> If a {KIE_SERVER} is not already configured in your {PRODUCT} environment, or if you require additional {KIE_SERVERS} in your {PRODUCT} environment, you must configure a {KIE_SERVER} to connect to {CENTRAL}.

Reviewer: ```suggestion
If a {KIE_SERVER} is not configured in your {PRODUCT} environment, or if you require additional {KIE_SERVERS} in your {PRODUCT} environment, you must configure a {KIE_SERVER} to connect to {CENTRAL}.
```

Author: Done

[NOTE]
====
If you are deploying {KIE_SERVER} on {OPENSHIFT}, see the {URL_DEPLOYING_ON_OPENSHIFT}#assembly-openshift-operator[_{DEPLOYING_OPENSHIFT_OPERATOR}_] document for instructions about configuring it to connect to {CENTRAL}.
====

ifeval::["{context}" == "execution-server"]

> {KIE_SERVER} can be managed or it can be unmanaged. If {KIE_SERVER} is unmanaged, you must manually create and maintain KIE containers (deployment units). If {KIE_SERVER} is managed, the {CONTROLLER} manages the {KIE_SERVER} configuration and you interact with the {CONTROLLER} to create and maintain KIE containers.

Reviewer: ```suggestion
{KIE_SERVER} can be managed or unmanaged. If {KIE_SERVER} is unmanaged, you must manually create and maintain KIE containers (deployment units). If {KIE_SERVER} is managed, the {CONTROLLER} manages the {KIE_SERVER} configuration and you interact with the {CONTROLLER} to create and maintain the KIE containers.
```

Author: Done

[NOTE]
====

> Only make the changes described in this section if {KIE_SERVER} will be managed by {CENTRAL} and you installed {PRODUCT} from the ZIP files.

Reviewer: This section contains various admonitions, I'd suggest changing this as normal text. 

```suggestion
Make the changes described in this section if {KIE_SERVER} is managed by {CENTRAL} and you have installed {PRODUCT} from the ZIP files.
```

Author: Done

> If you did not install {CENTRAL}, you can use the {HEADLESS_CONTROLLER} to manage {KIE_SERVER}, as described in xref:controller-con_{context}[].

Reviewer: ```suggestion
If you have installed {CENTRAL}, you can use the {HEADLESS_CONTROLLER} to manage {KIE_SERVER}, as described in xref:controller-con_{context}[].
```

Author: Done.

====
endif::[]

.Prerequisites
ifeval::["{context}" == "install-on-eap"]
* {CENTRAL} and {KIE_SERVER} are installed in the base directory of the {EAP} installation (`__EAP_HOME__`) as described in the following sections:
+
** <<eap-dm-install-proc_install-on-eap>>
** <<eap_execution_server_download_install_proc_install-on-eap>>

* Users with the following roles exist:
** In {CENTRAL}, a user with the role `rest-all`
** On the {KIE_SERVER}, a user with the role `kie-server`
+
For more information, see <<eap-users-create-proc_install-on-eap>>.
endif::[]

ifeval::["{context}" == "execution-server"]
* {CENTRAL} and {KIE_SERVER} are installed in the base directory of the {EAP} installation (`__EAP_HOME__`).

[NOTE]
====

> You should install {CENTRAL} and {KIE_SERVER} on different servers in production environments. In this sample situation, we use only one user named `controllerUser` that has both the `rest-all` and the `kie-server` roles. However, if you install {KIE_SERVER} and {CENTRAL} on the same server, for example in a development environment, make the changes described in this section in the shared `standalone-full.xml` file.

Reviewer: ```suggestion
You must install {CENTRAL} and {KIE_SERVER} on different servers in production environments. In this sample situation, we use only one user named `controllerUser`, containing both `rest-all` and the `kie-server` roles. However, if you install {KIE_SERVER} and {CENTRAL} on the same server, for example in a development environment, make the changes in the shared `standalone-full.xml` file as described in this section.
```

Author: Done

====

* Users with the following roles exist:
** In {CENTRAL}, a user with the role `rest-all`
** On the {KIE_SERVER}, a user with the role `kie-server`

endif::[]

ifeval::["{context}" == "packaging-deploying"]
* {CENTRAL} and {KIE_SERVER} are installed in the base directory of the {EAP} installation (`__EAP_HOME__`).

[NOTE]
====

> You should install {CENTRAL} and {KIE_SERVER} on different servers in production environments. In this sample situation, we use only one user named `controllerUser` that has both the `rest-all` and the `kie-server` roles. However, if you install {KIE_SERVER} and {CENTRAL} on the same server, for example in a development environment, make the changes described in this section in the shared `standalone-full.xml` file.

Reviewer: ```suggestion
You must install {CENTRAL} and {KIE_SERVER} on different servers in production environments. In this sample situation, we use only one user named `controllerUser`, containing both the `rest-all` and the `kie-server` roles. However, if you install {KIE_SERVER} and {CENTRAL} on the same server, for example in a development environment, make the changes in the shared `standalone-full.xml` file as described in this section.
```

Author: Done

| `org.kie.server.location`
| `\http://localhost:8080/kie-server/services/rest/server`
| The location of the {KIE_SERVER}. The URL for connecting to the API of {KIE_SERVER}.

|===

.JVM Properties for {CENTRAL} Instance
[cols="1,1,1a", options="header"]
|===
| Property
| Value
| Note

| `org.kie.server.user`
| `controllerUser`
| The user name with the role `kie-server`.

| `org.kie.server.pwd`
| `controllerUser1234;`
| The password of the user.

|===

Example for {KIE_SERVER} Instance:

[source,xml,subs="attributes+"]
----
<property name="org.kie.server.id" value="default-kie-server"/>
<property name="org.kie.server.controller" value="http://localhost:8080/{URL_COMPONENT_CENTRAL}/rest/controller"/>
<property name="org.kie.server.controller.user" value="controllerUser"/>
<property name="org.kie.server.controller.pwd" value="controllerUser1234;"/>
<property name="org.kie.server.location" value="http://localhost:8080/kie-server/services/rest/server"/>
----

Example for {CENTRAL} Instance:

[source,xml,subs="attributes+"]
----
<property name="org.kie.server.user" value="controllerUser"/>
<property name="org.kie.server.pwd" value="controllerUser1234;"/>
----

--
+

> . Verify the successful start of the {KIE_SERVER} by sending a GET request to `http://_SERVER:PORT_/kie-server/services/rest/server/`.

Reviewer: ```suggestion
. Send a GET request to `http://_SERVER:PORT_/kie-server/services/rest/server/` and verify the successful start of the {KIE_SERVER}.
```

Author: Done.

> Once authenticated, you get an XML response similar to this:

Reviewer: ```suggestion
Once authenticated, you receive an XML response similar to the following example:
```

Author: Done
